### PR TITLE
Rebase s3-proxy Docker image from "debian" to "amazonlinux"

### PR DESCRIPTION
**Pull request by dimaryaz**
This fixes vulnerabilities found by the ECR scanner.

Also upgrade nginx to 1.24.0.

### CODE REVIEW

File: s3-proxy/nginx.conf [hunk 1]
```diff
@@ -31,7 +31,7 @@ http {
             proxy_request_buffering off;
             proxy_ssl_verify on;
             proxy_ssl_verify_depth 5;
-            proxy_ssl_trusted_certificate /etc/nginx/certs.pem;
+            proxy_ssl_trusted_certificate /etc/ssl/certs/ca-bundle.trust.crt;
```
Comment (dimaryaz) [inline s3-proxy/nginx.conf:34]: This one is a change that I don't fully understand, and not even sure how to test properly.

We used to `cat /etc/ssl/certs/*.pem > /etc/nginx/certs.pem` - but the latest version of `amazonlinux` doesn't even have those files anymore. Instead, it has:

```
/etc/ssl/certs/ca-bundle.crt
/etc/ssl/certs/ca-bundle.trust.crt
```

I don't understand the difference; the only thing I can find is this: https://serverfault.com/questions/620003/difference-between-ca-bundle-crt-and-ca-bundle-trust-crt It implies that we should use `ca-bundle.crt`, unless we really care about security? But maybe we do?

Comment (dimaryaz) [reply s3-proxy/nginx.conf:34]: Oh, forgot that we potentially append custom certs to /etc/nginx/certs.pem. Need to undo this.

Comment (dimaryaz) [reply s3-proxy/nginx.conf:34]: Actually, amazonlinux:2023 only has one bundle, so that makes it easy...

**Review by nl0 — COMMENTED** [review]
seems fine, i don't see anything suspicious

**Review by sir-sigurd — APPROVED** [review]

**Pull request merged**